src/main/java/com/iabtcf/decoder/VendorsDecoder.java

```diff
@@ -0,0 +1,69 @@
+package com.iabtcf.decoder;
+
+import java.util.BitSet;
+import java.util.stream.IntStream;
+
+import static com.iabtcf.decoder.Field.Vendors.BIT_FIELD;
+import static com.iabtcf.decoder.Field.Vendors.END_VENDOR_ID;
+import static com.iabtcf.decoder.Field.Vendors.IS_A_RANGE;
+import static com.iabtcf.decoder.Field.Vendors.IS_RANGE_ENCODING;
+import static com.iabtcf.decoder.Field.Vendors.MAX_VENDOR_ID;
+import static com.iabtcf.decoder.Field.Vendors.NUM_ENTRIES;
+import static com.iabtcf.decoder.Field.Vendors.START_OR_ONLY_VENDOR_ID;
+
+/**
+ * @author SleimanJneidi
+ * @author evanwht1
+ */
+final class VendorsDecoder {
+
+	/**
+	 * Reads the Vendors fields from a TCF segment string. This can either be the entire OOB Disclosed Vendors or
+	 * OOB Allowed Vendors segments, or the Vendor Consent Section or Vendor Legitimate Interest Section parts of
+	 * the Core String.
+	 *
+	 * WARNING: This assumes you are at the correct position in the bit vector to read it's field. {@link TCModelDecoder}
+	 *          will have already read the {@link Field.Vendors#SEGMENT_TYPE} for the OOB segments and will have read up
```

Comment:
thats why I dont like BitVector being stateful and maintain a position pointer

Reply:
The decoders no longer have to correctly maintain a position for a specific subset of their fields, instead they just read from the bit vector much like you would an input stream. Since these bit vectors never live past the decoding step and aren't interacted with by the user, I think this is a good trade off since it cleans up the logic in the decoders that would otherwise have to create and maintain local position variables. I know speed isn't everything but I did test the other way and for a full string decode it was twice as slow.

Reply:
and why do you think having the position in BitVector makes it faster?

Reply:
Moving position into the bit vector requires less garbage to be created by the decoders when calculating offsets for variable length fields. Feel free to clone the benchmark repo or create your own to prove that (I tested that individual change a while ago but is not reflected in the results in the benchmark repo).
However, the bigger part for me is the readability improvement in the decoders by moving position to the bit vector.